learning_journal_basic1/tests.py

```diff
@@ -1,29 +1,162 @@
+def populated_db(request, sqlengine):
+    """Set up mock database for testing."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    with transaction.manager:
+        session.add(MyModel(title="Purgatory", body="The reality of pythonic purgatory cannot be denied.", date=datetime.datetime.utcnow()))
+
+    def teardown():
+        with transaction.manager:
+            session.query(MyModel).delete()
+
+    request.addfinalizer(teardown)
+
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """Testing a dummy request."""
+    return testing.DummyRequest(dbsession=new_session)
+
+
+def test_home_view(dummy_request, new_session):
+    """Test entries are in home view."""
+    from .views.default import home_view
+    new_session.add(MyModel(title="test", body="this is a test", date=datetime.strptime(entry['date'], TIME_FORMAT)))
+    new_session.flush()
+    info = home_view(dummy_request)
+    assert "entries" in info
+
+
+def test_detail_view(new_session):
+    """Test detail view has a body."""
+    from .views.default import detail_view
+    request = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyModel(title="test", body="this is a test", date=datetime.strptime(entry['date'], TIME_FORMAT)))
+    new_session.flush()
+    request.matchdict = {'id': '1'}
+    info = detail_view(request)
+    assert "this is a test" in info['entry'].body
+
+
+def test_entry_view():
+    """Test entry view."""
+    from .views.default import entry_view
+    request = testing.DummyRequest()
+    entry_view(request)
+    assert request.response.status_code == 200
+
+
+def test_edit_view(new_session):
+    """Test update view has a body."""
+    from .views.default import edit_view
+    request = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyModel(title="test", body="this is a test", date=datetime.strptime(entry['date'], TIME_FORMAT)))
+    new_session.flush()
+    request.matchdict = {'id': '1'}
+    info = edit_view(request)
+    assert info["entry"].body == "this is a test"
+
+# -------Functional Tests----------
+
+
+@pytest.fixture()
+def testapp(sqlengine):
+    """Setting Up TestApp."""
+    from learning_journal_basic1 import main
+    app = main({}, **DB_SETTINGS)
+    from webtest import TestApp
+    return TestApp(app)
+
+
+def test_layout_root_home(testapp, populated_db):
+    """Test layout root of home route."""
+    response = testapp.get('/', status=200)
+    assert b'Purgatory' in response.body
+
+
+def test_layout_root_entry(testapp):
+    """Test layout root of entry route."""
+    response = testapp.get('/journal/new_entry', status=200)
+    assert response.html.find("textarea")
+
+
+def test_layout_root_edit(testapp, populated_db):
+    """Test layout root of edit route."""
+    response = testapp.get('/journal/edit_view/{id:\d+}', status=200)
+    html = response.html
+    assert html.find("h2")
```

Comment:
would you care to make any assertions about what the content of the `<h2>` tag is?